pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -227,6 +227,24 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// predicate functions to filter the events based on the AWS resourceTags presence.
+	infraResourcePredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isAWSResourceTagsPresent(e.Object)
+		},
+		DeleteFunc: func(event.DeleteEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isAWSResourceTagsPresent(e.ObjectNew)
```

Comment:
This will work but is not an effective predicate. If you should check if the tags have changed between the new and old object.

Reply:
Addressed